### ISSUE TITLE
[generator] Resolve arrow references in permissions

### ISSUE THREAD

**Issue by swantzter** [swantzter/authzed-ts#2]
Basically, we need recursion.

When an arrow reference (`relation->relationOrPermission`) is encountered we need to look at the relation on the current object definition, and from that expand the types it references until we have reached all possible object types that can be the subject type.

Given this schema:

```zed
definition user {}

definition robot {}

definition lev1 {
    relation member: user | robot
}

definition lev2 {
    relation rel: lev1
    relation direct_member: user
    permission member = rel->member + user
}

definition lev3 {
    relation rel: lev2 | user
    permission member = rel->member
}
```

to resolve the `member` permission on `lev3` we need to look at the relation from the left hand side of the arrow expression: `rel`, which points to either a `lev2` or a `user`.
We must then try to find the right hand side of the arrow expression on either of these relations.

On `user` it is not found either as a permission or relation. Thus we discard this branch

On `lev2` however we find a permission named `member`, and, since it is a permission we must expand it.
It has a direct reference we can resolve to get our first path `{ objectType: 'lev2', relation: 'direct_member' }` and another arrow expression we must expand the same way we just did. We look at the relation defined on `lev2` that's references on the right hand side of this new arrow expression: `rel`, which points to a `lev1`

On `lev1` we try to find a permission named `member`, we don't, so we try to find a relation named `member` and we do! this means our search is complete and we get our second path: `{ objectType: 'lev1', relation: 'member' }`

these two relations gets added as the resulting array at the TODO marker here:

https://github.com/swantzter/authzed-ts/blob/10b9a36b592c7002f909c17b14e24a1931d8621a/generator/lib/generator.ts#L116-L125

basically: `currentObj.permissions[permission.name] ??= resolveArrowReference(whateverArgsAreNeeded)`
